extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java

```diff
@@ -259,20 +247,36 @@ private void applyGlobalConfig(Session session, KubernetesConfig config) {
      * @param config The {@link PlatformConfiguration} instance
      * @param now
      */
-    private void applyConfig(Session session, String target, String name, PlatformConfiguration config, ZonedDateTime now) {
+    private void applyConfig(Session session, Project project, String target, String name, PlatformConfiguration config,
+            ZonedDateTime now) {
         //Labels
         config.getLabels().forEach((key, value) -> {
             session.resources().decorate(target, new AddLabelDecorator(new Label(key, value)));
         });
 
+        if (OPENSHIFT.equals(target)) {
+            session.resources().decorate(OPENSHIFT, new AddLabelDecorator(new Label(OPENSHIFT_APP_RUNTIME, QUARKUS)));
+        }
+
         //Annotations
         config.getAnnotations().forEach((key, value) -> {
             session.resources().decorate(target, new AddAnnotationDecorator(new Annotation(key, value)));
         });
+
+        ScmInfo scm = project.getScmInfo();
+        String vcsUrl = scm != null ? scm.getUrl() : Annotations.UNKNOWN;
+        String commitId = scm != null ? scm.getCommit() : Annotations.UNKNOWN;
+
+        session.resources().decorate(target, new RemoveAnnotationDecorator(Annotations.VCS_URL));
+        session.resources().decorate(target, new RemoveAnnotationDecorator(Annotations.COMMIT_ID));
```

Comment:
Can we have a comment here please explaining that `dekorate` adds these automatically and we want to replace them with more specific ones?